packages/tab/src/index.d.ts

```diff
@@ -0,0 +1,44 @@
+import { FunctionComponent, ReactChild, Component, MouseEvent } from 'react'
+
+interface Styling {
+    css?: Record<string, string | number>
+    style?: Record<string, string | number>
+}
+
+interface ListProps {
+    className?: string
+}
+
+type List = FunctionComponent<ListProps & Styling>
+
+interface ListItemProps {
+    id: string | number
+    active?: boolean
+    children?: string
+    className?: string
+    onClick?: (i: number, event: MouseEvent) => void
+}
+interface ListItemAnchorProps {
+    href: string
+    innerRef?: (element: HTMLAnchorElement | null) => void
+}
+interface ListItemButtonProps {
+    innerRef?: (element: HTMLButtonElement | null) => void
+}
+
+type ListItem = FunctionComponent<ListItemProps & (ListItemAnchorProps | ListItemButtonProps) & Styling>
+
+interface PanelProps {
+    className?: string
+    labelledBy: string | number
+}
+
+type Panel = FunctionComponent<PanelProps & Styling>
+
+declare const exports: {
```

Comment:
This is me not understanding TS.  Why `declare` here (as well as export later)?

Reply:
`declare` here is a way of declaring the type for an object that is created on the JS side.

Then the `export default` is saying that that object is the default export in the JS.

More here:
https://stackoverflow.com/a/41749803
https://www.typescriptlang.org/docs/handbook/declaration-files/by-example.html